mantidproject/mantid
# (Topic Branches) Add Recovery to config and remove WS from ADS after

**Description of work.**
- Wraps all the new code in try catch blocks to exit the thread if something goes wrong
- Adds a new config option for these changes
- Adds WS blacklisting, removing all workspaces that did not exist prior
- Automatically saves in timestamped folders, which are rotated so only 5 are present (by default)

**To test:**
- Check `%appdata%/recovery` a maximum of 5 checkpoints should be present
- It should save every minute
- Switch the config option off and check that saving stops
- Add a failure to the python script (e.g. `str(1)+int(2)`) check it stops once

---

#### Reviewer ####

Please comment on the following ([full description](http://www.mantidproject.org/Individual_Ticket_Testing)):

##### Code Review #####


##### Functional Tests #####


Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.
